safe-global/safe-wallet-web
# feat: Add remove delegate option and adjust delegate list layout

## What it solves

Resolves [SW-285](https://www.notion.so/safe-global/Extend-Proposers-section-by-add-remove-11c8180fe5738090b08bca60dbea5384)

## ToDos

- [x] Implement design
- [x] Dynamically update lists when adding/removing delegate
- [x] Show notification when adding/removing proposers

## How this PR fixes it

- Adds a remove button next to each delegate in the list
- Changes the list to a table for consistency
- A short dialog to confirm deletion
- Signs the typed data and sends a request to the CGW
- Renames delegates to proposers

## How to test it

1. Go to a Safe that has a delegate added
2. Go the the settings -> setup page
3. Observe a button next to the delegate to delete
4. Press the button and observe a dialog opening
5. Press Delete in the dialog
6. Observe the delegate is deleted
7. There should be evens when pressing the Add/Delete button, Canceling the dialog and submitting the form
8. There should be notifications when submitting add/delete

## Screenshots
<img width="1293" alt="Screenshot 2024-10-23 at 15 36 47" src="https://github.com/user-attachments/assets/13442c5c-820e-413f-b1d2-3d73c76e69e0">
<img width="653" alt="Screenshot 2024-10-23 at 15 37 03" src="https://github.com/user-attachments/assets/13cd4bd8-0a73-49bf-b66a-84895447f56c">
<img width="663" alt="Screenshot 2024-10-23 at 15 37 12" src="https://github.com/user-attachments/assets/31a05966-6e98-439d-a681-352706fe93ae">
<img width="364" alt="Screenshot 2024-10-23 at 15 34 45" src="https://github.com/user-attachments/assets/3095d4b1-d5e8-4362-a067-169244699ddc">
<img width="360" alt="Screenshot 2024-10-23 at 15 32 25" src="https://github.com/user-attachments/assets/de134cbf-b217-472e-92bb-019b2de45823">

## Checklist
